src/nvim/normal.c

```diff
@@ -5924,6 +5924,9 @@ static void nv_replace(cmdarg_T *cap)
     curwin->w_set_curswant = true;
     set_last_insert(cap->nchar);
   }
+
+  foldUpdateAll(curwin);
+  foldOpenCursor();
```

Comment:
Why is this needed? `insert_enter ()` should be enough. The comment above `edit()` mentions that it handles `gr`, `gR`, `R`, etc.